Weather/Weather/Repositories/SearchRepository.swift

```diff
@@ -7,3 +7,26 @@
 //
 
 import Foundation
+
+class SearchRepository: NSObject {
+    private let api = APIService.share
+    
+    override init() {
+        super.init()
+    }
+    
+    func searchLocation(keyword: String, completion: @escaping (BaseResult<SearchResponse>) -> Void ) {
+        let searchRequest = SearchRequest(keyword: keyword)
+        api.request(input: searchRequest) { (object: SearchResponse?, error) in
```

Comment:
```suggestion
        api.request(input: searchRequest) { (response: SearchResponse?, error) in
            guard let searchResponse = response else {
                return completion(.failed(error: error))
            }
            switch searchResponse.status {
            case Constants.ok:
                completion(.success(searchResponse))
            default:
                completion(.failed(error: .googleError(searchResponse.status)))
            }
        }
```
đặt rõ tên biến, không đặt kiểu mơ hồ như "object"